docs/source/llms/tracing/index.rst

```diff
@@ -0,0 +1,558 @@
+
+            return response
+
+
+    invocation(16)
+
+The screenshot below shows the wrapped function call having its inputs and output captured from within a span.
+
+.. figure:: ../../_static/images/llms/tracing/external-function.png
+    :alt: External Function tracing
+    :width: 100%
+    :align: center
+
+
+Tracing Client APIs
+-------------------
+
+The MLflow client API provides a comprehensive set of thread-safe methods for manually managing traces. These APIs allow for fine-grained 
+control over tracing, enabling you to create, manipulate, and retrieve traces programmatically. This section will cover how to use these APIs 
+to manually trace a model, providing step-by-step instructions and examples.
+
+Starting a Trace
+^^^^^^^^^^^^^^^^
+
+Unlike with the fluent API, the MLflow Trace Client API requires that you explicitly start a trace before adding child spans. This initial API call 
+starts the root span for the trace, providing a context request_id that is used for associating subsequent spans to the root span. 
+
+To start a new trace, use the :py:meth:`mlflow.client.MlflowClient.start_trace` method. This method creates a new trace and returns the root span object.
+
+.. code-block:: python
+
+    from mlflow import MlflowClient
+
+    client = MlflowClient()
+
+    # Start a new trace
+    root_span = client.start_trace("my_trace")
+
+    # The request_id is used for creating additional spans that have a hierarchical association to this root span
+    request_id = root_span.request_id
+
+Adding a Child Span
+^^^^^^^^^^^^^^^^^^^
+
+Once a trace is started, you can add child spans to it. Child spans allow you to break down the trace into smaller, more manageable segments, 
+each representing a specific operation or step within the overall process.
+
+.. code-block:: python
+
+    # Create a child span
+    child_span = client.start_span(
+        name="child_span",
+        request_id=request_id,
+        parent_id=root_span.span_id,
+        inputs={"input_key": "input_value"},
+        attributes={"attribute_key": "attribute_value"},
+    )
+
+Ending a Span
+^^^^^^^^^^^^^
+
+After performing the operations associated with a span, you must end the span explicitly using the ``end_span`` method. Make note of the two required fields 
+that are in the API signature:
+
+- **request_id**: The identifier associated with the root span
+- **span_id**: The identifier associated with the span that is being ended
+
+In order to effectively end a particular span, both the root span (returned from calling ``start_trace``) and the targeted span (returned from calling ``start_span``)
+need to be identified when calling the ``end_span`` API. It's a good practice to ensure that the root span's ``request_id`` is readily accessible from all locations
+that you will be starting or ending spans to ensure that any spans created within the context of a trace are properly associated with the root span. 
```

Comment:
The `request_id` can be accessed from any children spans' `request_id` property e.g. `client.end_span(request_id=child_span.request_id`, ...)`, not only root span. I don't think we should promote passing the root span or request ID around their code.

Reply:
Removed the sentence that refers to this